technique/reflectometry/base.py

```diff
@@ -255,6 +255,59 @@ def slit_check(theta, footprint, resolution):
     print("s2vg={}".format(s2))
 
 
+def auto_height(laser_block: str, fine_height_block: str, target: float = 0.0, continue_if_NaN: bool=False):
+    """
+    Moves the sample fine height axis so that it is centred on the beam, based on the readout of a laser height gun.
+
+    Args:
+        laser_block: The block for the laser offset from centre
+        fine_height_block: The block for the sample fine height
+        target: The target laser offset
+        continue_if_NaN: Defines what to do in case of invalid values. If True, ignore errors and continue execution.
+            If False, break and wait for user input. (default: False)
+        
+        >>> auto_height(b.KEYENCE, b.HEIGHT2)
+
+        Moves HEIGHT2 by (KEYENCE * (-1))
+
+        >>> auto_height(b.KEYENCE, b.HEIGHT2, target=0.5, continue_if_NaN=True)
+
+        Moves HEIGHT2 by (target - b.KEYENCE) and does not interrupt script execution if an invalid value is read.
+    """
+    try:
+        current_laser_offset = g.cget(laser_block)["value"]
```

Comment:
This function is quite long, could we pull the calculating of the target_height out into its own function e.g. _auto_heigh_calculate_target_height?